src/System Application/App/Document Processor/src/PDFHelperImpl.Codeunit.al

```diff
@@ -0,0 +1,68 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace System.IO;
+
+using System.Utilities;
+using System;
+
+/// <summary>
+/// Codeunit that provides helper functions for PDF processing.
+/// </summary>
+codeunit 3109 "PDF Helper Impl"
+{
+    Access = Internal;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    var
+        UnsupportedImageFormatErr: Label 'Unsupported image format: %1', Comment = '%1 is the image format that is not supported.';
+
+    procedure ConvertPdfToImage(DocumentStream: InStream; var ImageStream: InStream; ImageFormat: Enum "Image Format"; PageNumber: Integer): Boolean
+    var
+        PdfConverterInstance: DotNet PdfConverter;
+        PdfTargetDevice: DotNet PdfTargetDevice;
+        MemoryStream: DotNet MemoryStream;
+        ImageMemoryStream: DotNet MemoryStream;
+        SharedDocumentStream: InStream;
+    begin
+        // Empty stream, no actions possible on the stream so return immediately
+        if DocumentStream.Length < 1 then
+            exit(false);
+
+        // Use a shared stream and reset the read pointer to beginning of stream.
+        SharedDocumentStream := DocumentStream;
+        if SharedDocumentStream.Position > 1 then
+            SharedDocumentStream.ResetPosition();
+
+        MemoryStream := MemoryStream.MemoryStream();
+        MemoryStream := SharedDocumentStream;
```

Comment:
There were big issues with direct assigments of memory streams, we have removed them from the product. I cannot remember why exactly.

I believe CopyStream is better. Example in codeunit 3009 DotNet_MemoryStream, we also use it in the other places.